### PR TITLE
New Extention: SSH-Tunnel-Manager

### PR DESCRIPTION
**Pull request by MinatoHikari**
## Description

New Extention: SSH-Tunnel-Manager
Manage SSH tunnels with system ssh command

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

**Comment by pernielsentikaer** [comment]
Hi @MinatoHikari 👋 

Did you consider building it into https://www.raycast.com/dimagrossman/ssh-manager

**Comment by MinatoHikari** [comment]
> Hi @MinatoHikari 👋
> 
> Did you consider building it into https://www.raycast.com/dimagrossman/ssh-manager

Emm, never considered it before, If I want to connect to my servers, i will use tabby or other clients.
But port forword is usually an unique request, like linking to database which won't allow direct remote connection.
I use [this tool](https://www.tynsoe.org/stm/) before, but it doesn't have apple silicon support. And I want find something to write to publish my first extension, So I build this extension.
Do you think it is better to merge it into ssh-manager?

**Comment by MinatoHikari** [comment]
@pernielsentikaer

File: extensions/ssh-tunnel-manager/src/screens/FormScreen.tsx [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+            onSubmit={(values: Values) => {
+              shouldEstablish.current = false;
+              return handleSubmit(values);
+            }}
+          />
+          <Action
+            title="Tunnel List"
+            onAction={pop}
+            shortcut={{
+              modifiers: ["shift"],
```
Comment (pernielsentikaer) [inline extensions/ssh-tunnel-manager/src/screens/FormScreen.tsx:40]: Can we use something else? `⌘+tab` is used in the form to navigate one field up

File: extensions/ssh-tunnel-manager/src/screens/FormScreen.tsx [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+      onSubmit({ ...values });
+    }
+    pop();
+  });
+
+  return (
+    <Form
+      actions={
+        <ActionPanel>
+          <Action.SubmitForm title="Create and establish tunnel" onSubmit={handleSubmit} />
```
Comment (pernielsentikaer) [inline extensions/ssh-tunnel-manager/src/screens/FormScreen.tsx:28]:
```suggestion
          <Action.SubmitForm title="Create And Establish Tunnel" onSubmit={handleSubmit} />
```

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi 👋 

Thanks for your contribution 💪 

I have now tested your extension, and I have some feedback ready for you 😊

I'm looking forward to testing this extension again 🔥 

Request a new review when you are ready. Feel free to contact me here or at [Slack](https://www.raycast.com/community) if you have any questions.

File: extensions/ssh-tunnel-manager/src/screens/FormScreen.tsx (hunk 1 above)
Comment (MinatoHikari) [reply extensions/ssh-tunnel-manager/src/screens/FormScreen.tsx]: change shortcut to opt+shift+tab

**Comment by MinatoHikari** [comment]
@pernielsentikaer Hello, suggestions were applied. Request a new review here.

File: extensions/ssh-tunnel-manager/src/screens/FormScreen.tsx [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+              return handleSubmit(values);
+            }}
+          />
+          <Action
+            title="Tunnel List"
+            onAction={pop}
+            shortcut={{
+              modifiers: ["opt", "shift"],
+              key: "tab",
+            }}
```
Comment (pernielsentikaer) [inline extensions/ssh-tunnel-manager/src/screens/FormScreen.tsx:42]: Did you consider ⌘+n instead?

Comment (MinatoHikari) [reply extensions/ssh-tunnel-manager/src/screens/FormScreen.tsx:42]: fine, let me change it

File: extensions/ssh-tunnel-manager/src/screens/ListScreen.tsx [hunk 4]
```diff
@@ -0,0 +1,203 @@
...
+      isLoading={isLoading}
+      isShowingDetail={listData.length !== 0}
+      searchBarPlaceholder={"Search Tunnels"}
+      onSelectionChange={(id) => {
+        setSelectedId(id);
+      }}
+      actions={
+        <ActionPanel>
+          <Action title="Refresh List" onAction={refreshList} />
+          <Action title="Create Tunnel" onAction={toCreate} />
```
Comment (pernielsentikaer) [inline extensions/ssh-tunnel-manager/src/screens/ListScreen.tsx:102]: Double of same action?

File: extensions/ssh-tunnel-manager/src/screens/ListScreen.tsx [hunk 5]
```diff
@@ -0,0 +1,203 @@
...
+                    onAction={refreshList}
+                    title="Refresh List"
+                    shortcut={{
+                      modifiers: ["shift", "cmd"],
+                      key: "r",
+                    }}
+                  />
+                  <Action
+                    onAction={deleteTunnel}
+                    title="Delet Tunnel"
```
Comment (pernielsentikaer) [inline extensions/ssh-tunnel-manager/src/screens/ListScreen.tsx:163]:
```suggestion
                    title="Delete Tunnel"
                    style={Action.Style.Destructive}
```

File: extensions/ssh-tunnel-manager/src/screens/ListScreen.tsx (hunk 4 above)
Comment (MinatoHikari) [reply extensions/ssh-tunnel-manager/src/screens/ListScreen.tsx:102]: emm, i want to use same shortcut whenever list is empty or not, do you have any suggestions?

Comment (pernielsentikaer) [reply extensions/ssh-tunnel-manager/src/screens/ListScreen.tsx:102]: I was talking about this

Raycast-EHBJvKjs@2x

Raycast-axxc0DqP@2x

Comment (pernielsentikaer) [reply extensions/ssh-tunnel-manager/src/screens/ListScreen.tsx:102]: same for refresh 😊

Comment (MinatoHikari) [reply extensions/ssh-tunnel-manager/src/screens/ListScreen.tsx:102]: I know, but i don't know how to override the default shortcut enter and ⌘+enter

Comment (pernielsentikaer) [reply extensions/ssh-tunnel-manager/src/screens/ListScreen.tsx:102]: The bind should work but it won't show it for the first two items - that's how it works - the first two items will show enter and ⌘+enter

Comment (MinatoHikari) [reply extensions/ssh-tunnel-manager/src/screens/ListScreen.tsx]: ok, deleted

**Review by pernielsentikaer — APPROVED** [review]
Hi 👋 

Looks good to me, approved 🔥

**Pull request merged**